File: app/src/main/java/ie/dublinbuspal/android/data/local/dao/RouteDao.java
package ie.dublinbuspal.android.data.local.dao;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Transaction;
import ie.dublinbuspal.android.data.local.DbInfo;
import ie.dublinbuspal.android.data.local.entity.Route;

import java.util.List;

@Dao
public abstract class RouteDao implements BaseDao<Route> {

    @Query(DbInfo.SELECT_ROUTES)
    public abstract List<Route> selectAll();

    @Query(DbInfo.DELETE_ROUTES)
    public abstract void deleteAll();

    @Transaction
    public void replaceAll(List<Route> routes) {
        deleteAll();
        insertAll(routes);
    }

}

File: data/src/main/kotlin/ie/dublinbuspal/data/dao/RouteDao.kt
package ie.dublinbuspal.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import ie.dublinbuspal.data.entity.RouteEntity
import io.reactivex.Maybe

@Dao
interface RouteDao {

    @Query("SELECT * FROM routes")
    fun selectAll(): Maybe<List<RouteEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(entities: List<RouteEntity>)

    @Query("DELETE FROM routes")
    fun deleteAll()

}
